src/mscorlib/corefx/System/Globalization/TextInfo.Unix.cs

```diff
@@ -36,9 +36,35 @@ private unsafe string ChangeCase(string s, bool toUpper)
             }
 
             string result = string.FastAllocateString(s.Length);
-            fixed (char* pBuf = result)
+
+            fixed (char* pResult = result)
             {
-                Interop.GlobalizationInterop.ChangeCase(s, s.Length, pBuf, result.Length, toUpper, m_needsTurkishCasing);
+                if (IsAsciiCasingSameAsInvariant && s.IsAscii())
```

Comment:
I would rather do it this way, IsAsciiCasingSameAsInvariant should be cached per TextInfo, which are not created frequently.  Strings are, and I would rather us save the walk of the string if it going to be worthless.  Just my two cents.